drivers/nfs.py

```diff
@@ -186,38 +194,74 @@ def unmount(mountpoint, rmmountpoint):
-        textnode = dom.createTextNode(target)
-        subentry.appendChild(textnode)
-
-        # Access is not always provided by showmount return
-        # If none is provided we need to assume "*"
-        array = val.split()
-        path = array[0]
-        access = array[1] if len(array) >= 2 else "*"
-        subentry = dom.createElement("Path")
-        entry.appendChild(subentry)
-        textnode = dom.createTextNode(path)
-        subentry.appendChild(textnode)
+    fail = True
+    try:
+        cmd = [SHOWMOUNT_BIN, "--no-headers", "-e", target]
+        for val in util.pread2(cmd).split('\n'):
+            if not len(val):
+                continue
+            entry = dom.createElement('Export')
+            element.appendChild(entry)
+
+            subentry = dom.createElement("Target")
+            entry.appendChild(subentry)
+            textnode = dom.createTextNode(target)
+            subentry.appendChild(textnode)
+
+            # Access is not always provided by showmount return
+            # If none is provided we need to assume "*"
+            array = val.split()
+            path = array[0]
+            access = array[1] if len(array) >= 2 else "*"
+            subentry = dom.createElement("Path")
+            entry.appendChild(subentry)
+            textnode = dom.createTextNode(path)
+            subentry.appendChild(textnode)
+
+            subentry = dom.createElement("Accesslist")
+            entry.appendChild(subentry)
+            textnode = dom.createTextNode(access)
+            subentry.appendChild(textnode)
+        fail = False
+    except Exception:
+        util.SMlog("Unable to scan exports with rpcingo, trying NFSv4")
 
-        subentry = dom.createElement("Accesslist")
-        entry.appendChild(subentry)
-        textnode = dom.createTextNode(access)
-        subentry.appendChild(textnode)
+    try:
+        mountpoint = "%s/%s" % NFS4_TMP_MOUNTPOINT, target
+        soft_mount(mountpoint, target, NSFv4_PSEUDOFS, transport, nfsversion='4')
+        paths = util.pread2([LS])
+        unmount(mountpoint, NSFv4_PSEUDOFS)
+        for path in paths.split("\n"):
+            entry = dom.createElement('Export')
+            element.appendChild(entry)
+
+            subentry = dom.createElement("Target")
+            entry.appendChild(subentry)
+            textnode = dom.createTextNode(target)
+            subentry.appendChild(textnode)
+            subentry = dom.createElement("Path")
+            entry.appendChild(subentry)
+            textnode = dom.createTextNode(path)
+            subentry.appendChild(textnode)
+
+            subentry = dom.createElement("Accesslist")
+            entry.appendChild(subentry)
+            # Assume everyone as we do not have any info about it
+            textnode = dom.createTextNode("*")
+            subentry.appendChild(textnode)
+        fail = False
+    except Exception:
+        util.SMlog("Unable to scan exports with NFSv4 pseudo FS mount")
```

Comment:
AFAIK you only need to do this if showmount doesn't give any results. Otherwise, you already have the exports for both v3 and v4.